vortex-array/src/view.rs

```diff
@@ -112,25 +113,37 @@ impl ArrayView {
     }
 
     // TODO(ngates): should we separate self and DType lifetimes? Should DType be cloned?
-    pub fn child(&self, idx: usize, dtype: &DType, len: usize) -> Option<Self> {
-        let child = self.array_child(idx)?;
+    pub fn child(&self, idx: usize, dtype: &DType, len: usize) -> VortexResult<Self> {
+        let child = self
+            .array_child(idx)
+            .ok_or_else(|| vortex_err!("ArrayView: array_child({idx}) not found"))?;
         let flatbuffer_loc = child._tab.loc();
 
-        let encoding = self.ctx.lookup_encoding(child.encoding())?;
+        let encoding = self
+            .ctx
+            .lookup_encoding(child.encoding())
+            .unwrap_or_else(|| {
+                // We must return an EncodingRef, which requires a static reference.
+                // OpaqueEncoding however must be created dynamically, since we do not know ahead
+                // of time which of the ~65,000 unknown code IDs we will end up seeing. Thus, we
+                // allocate (and leak) 2 bytes of memory to create a new encoding.
+                Box::leak(Box::new(OpaqueEncoding(child.encoding())))
```

Comment:
Thinking out loud, the alternative is to have a lazy_static hash map of OpaqueEncodings by code... which for LOTS of Opaque encodings would mean not leaking so much

Reply:
Yes that was my thought as well. But you'd need one for every possible u16 value, so the break-even point is having 65,536 arrays with unknown encoding.

Reply:
So if you have many many columns, and many thousands of chunks with unknown encodings, it could be worse. VS the static map is 128KiB of fixed overhead, but the vast majority of the time you won't need it and it's just taking up space in the binary needlessly

Reply:
I think this will really only be a problem in a long lived proxy but I guess we can fix it when we get there?

Reply:
Filed https://github.com/spiraldb/vortex/issues/905 to reconsider this at a later date